### ISSUE TITLE
Better version reporting in Visual Studio for missing or mismatched CUDA Toolkit version

### ISSUE THREAD

**Issue by int19h** [NVIDIA/cuda-samples#268]
When trying to open sample projects using the freshly released CUDA Toolkit 12.5 in Visual Studio 2022, they do open with no files listed in Solution Explorer, and an error message in the Output pane as follows:

> error : Designtime build failed for project 'C:\git\NVIDIA\cuda-samples\Samples\0_Introduction\matrixMul\matrixMul_vs2022.vcxproj' configuration 'Debug|x64'. IntelliSense might be unavailable.

Build is similarly broken - it just silently fails, no additional error messages.

After some investigation I found out that this is happening because the .vcxproj file contains the following:
```xml
    <Import Project="$(CUDAPropsPath)\CUDA 12.4.props" />
    ...
    <Import Project="$(CUDAPropsPath)\CUDA 12.4.targets" />
```
Since I only have 12.5 installed, and thus only its .props and .targets are present, these imports fail with the non-descriptive error message as described above. This is unfortunate, given that new CUDA releases are a regular occasion and thus this is an issue that is bound to come up regularly. Therefore, it would be good to provide better diagnostics.

This can be done, firstly, by making the import itself conditional on the existence of the imported file:
```xml
    <Import Project="$(CUDAPropsPath)\CUDA 12.4.props" Condition="Exists('$(CUDAPropsPath)\CUDA 12.4.props')" />
    <Import Project="$(CUDAPropsPath)\CUDA 12.4.targets" Condition="Exists('$(CUDAPropsPath)\CUDA 12.4.targets')" />
```

With this change alone, the project will now load and build without any errors at all (but also without the correct output). The following can then be added to produce a clear error message when attempting to build the project:
```xml
  <Target Name="CheckCudaToolkitVersion" BeforeTargets="PrepareForBuild">
    <Error
      Condition="!Exists('$(CUDAPropsPath)\CUDA 12.4.props') Or !Exists('$(CUDAPropsPath)\CUDA 12.4.targets')"
      Text="This project requires CUDA Toolkit 12.4, which can be downloaded from https://developer.nvidia.com/cuda-12-4-1-download-archive. Other CUDA Toolkit versions are not supported." />
  </Target> 
```
End result looks like this:
![image](https://github.com/NVIDIA/cuda-samples/assets/1175142/8d221451-9085-4b43-91e0-fa1ae28a1c89)

Better yet would be to place the target toolkit version in a separate `<Property>` and then reference that throughout the .vcxproj, so that it only needs to be updated in one place for new CUDA Toolkit versions.

**Comment by iamparik** [comment]
Agreed @int19h, I was caught off-guard with this error for all sample codes as well- would have appreciated better error message. Saw another issue from 2023 when some folks had a similar issue for an older version i believe. 

Anyways, a quick fix for others here to find a workaround for now would be to change the .vcxproj file with references to 12.4 CUDA versions to 12.5 or the CUDA version they have installed. So for 12.5 version it would be:

    <Import Project="$(CUDAPropsPath)\CUDA 12.5.props" />
    ...
    <Import Project="$(CUDAPropsPath)\CUDA 12.5.targets" />

**Comment by nbowling** [comment]
I too was struggling with this and have suggested editing the proj file to reflect the new props file. However, manually editing proj files is fraught with potential side effects so I decided to do the thing that I hadn't done properly which was to RTFM. I found in the NVIDIA dos the following.

"Open the Visual Studio project, right click on the project name, and select BUILD DEPENDENCIES > BUILD CUSTOMIZATIONS... then select the CUDA Toolkit version you would like to target."

ref https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html
Scroll down to section 4.2

Boom! everything compiles albeit with some warnings about conflicting libraries.

**Comment by rwarmstr** [comment]
We've switched the repository over to CMake to generate artifacts for different build systems including Visual Studio solution files; this should resolve the issue.
